### PR TITLE
Add Thetis Market

### PR DESCRIPTION
**Pull request by thetis-market**

### CODE REVIEW

**Comment by llamatester** [comment]
The thetis-market-aggregator adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators thetis-market-aggregator
    
    🦙 Running THETIS-MARKET-AGGREGATOR adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 8.14 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs thetis-market
    
    🦙 Running THETIS-MARKET adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    Version -> THETIS-MARKET-DERIVATIVE
    ---------
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 5.02 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)
    
    
    Version -> THETIS-MARKET-AGGREGATOR
    ---------
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 8.14 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees thetis-market.ts
    
    🦙 Running THETIS-MARKET.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Total fees: 3.49 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Total supply side revenue: 2.79 k
    Daily fees: 2.13 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Daily supply side revenue: 1.70 k
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

File: dexs/thetis-market/index.ts
Comment (g1nt0ki) [inline dexs/thetis-market/index.ts:1]: thanks for the PR.

This file can be skipped/deleted. We will make two listings, one for aggregator & one for derivatives

File: aggregators/thetis-market-aggregator/index.ts
Comment (g1nt0ki) [inline aggregators/thetis-market-aggregator/index.ts:1]: I think this should be moved to the `aggregator-derivatives` folder

File: dexs/thetis-market/derivative/index.ts
Comment (g1nt0ki) [inline dexs/thetis-market/derivative/index.ts:1]: can be moved one level up since the other file is gonna be removed

File: fees/thetis-market.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+const fetch = async (timestamp: number) => {
+  const startTime = getUniqStartOfTodayTimestamp(new Date(timestamp * 1000));
+  const endTime = startTime + 86400;
+
+  const feeRes = await fetchURL(
+    `${endpoint}fee-daily?startTime=${startTime}&endTime=${endTime}`
+  );
+
+  if (feeRes.length) {
+    const dailyFees = new BigNumber(feeRes[0].total).dividedBy(1e18);
```
Comment (g1nt0ki) [inline fees/thetis-market.ts]: feel like bignumber is overkill here :)

File: fees/thetis-market.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const totalFees = new BigNumber(feeRes[0].cumulative).dividedBy(1e18);
+
+    return {
+      totalFees: totalFees.toString(),
+      totalSupplySideRevenue: totalFees.times(80).dividedBy(100).toString(),
+      dailyFees: dailyFees.toString(),
+      dailySupplySideRevenue: dailyFees.times(80).dividedBy(100).toString(),
```
Comment (g1nt0ki) [inline fees/thetis-market.ts]: can also track dailyRevenue as `fees * 0.2`? Also, it is in number format, no need to do toString, not sure, maybe BigNumber is also a valid response format.

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
mostly good, again, thanks for the PR

File: dexs/thetis-market/index.ts
Comment (thetis-market) [reply dexs/thetis-market/index.ts:1]: Thanks, do you need me to update again?

File: aggregators/thetis-market-aggregator/index.ts
Comment (thetis-market) [reply aggregators/thetis-market-aggregator/index.ts:1]: Currently Thetis Market has 2 main products: swap-aggregator and derivative. That's why we put them in 2 different folders.

**Comment by llamatester** [comment]
The thetis-market-aggregator adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators thetis-market-aggregator
    
    🦙 Running THETIS-MARKET-AGGREGATOR adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 8.14 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs thetis-market
    
    🦙 Running THETIS-MARKET adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    Version -> THETIS-MARKET-DERIVATIVE
    ---------
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 5.02 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)
    
    
    Version -> THETIS-MARKET-AGGREGATOR
    ---------
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 8.14 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees thetis-market.ts
    
    🦙 Running THETIS-MARKET.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Total fees: 3.49 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Total revenue: 2.79 k
    └─ Methodology: Revenue is 80% of the total fees, which are distributed to LP stakers
    Daily fees: 2.13 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Daily revenue: 1.70 k
    └─ Methodology: Revenue is 80% of the total fees, which are distributed to LP stakers
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

File: fees/thetis-market.ts (hunk 2 above)
Comment (thetis-market) [reply fees/thetis-market.ts]: We have updated, revenue = 80% fees

Comment (g1nt0ki) [reply fees/thetis-market.ts]: no, the update has a bug, old code is correct: 
image
https://docs.llama.fi/list-your-project/other-dashboards/dimensions

**Comment by llamatester** [comment]
The thetis-market-aggregator adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators thetis-market-aggregator
    
    🦙 Running THETIS-MARKET-AGGREGATOR adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 8.14 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs thetis-market
    
    🦙 Running THETIS-MARKET adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    Version -> THETIS-MARKET-DERIVATIVE
    ---------
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 5.02 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)
    
    
    Version -> THETIS-MARKET-AGGREGATOR
    ---------
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 8.14 M
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees thetis-market.ts
    
    🦙 Running THETIS-MARKET.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 06 Dec 2024 00:00:00 GMT
    End Date:	Sat, 07 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Total fees: 3.49 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Total supply side revenue: 2.79 k
    └─ Methodology: SupplySideRevenue is 80% of the total fees, which are distributed to LP stakers
    Total revenue: 697
    └─ Methodology: Revenue is 20% of the total fees
    Daily fees: 2.13 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Daily supply side revenue: 1.70 k
    └─ Methodology: SupplySideRevenue is 80% of the total fees, which are distributed to LP stakers
    Daily revenue: 425
    └─ Methodology: Revenue is 20% of the total fees
    End timestamp: 1733443200 (2024-12-06T00:00:00.000Z)

File: fees/thetis-market.ts (hunk 2 above)
Comment (thetis-market) [reply fees/thetis-market.ts]: Yes, I have reverted the code.

File: aggregators/thetis-market-aggregator/index.ts
Comment (g1nt0ki) [inline aggregators/thetis-market-aggregator/index.ts:1]: sorry, I missed this part, I assumed the aggregator was dex aggregator (of multiple dexes) or derivatives aggregator, not your own dex & derivatives product. is that the case here?

Comment (thetis-market) [reply aggregators/thetis-market-aggregator/index.ts:1]: No, Thetis owns the derivatives product (we have trading pools). Additionally, we offer an independent product called the swap aggregator, which can utilize Thetis pools or other DEXs within the Aptos ecosystem.

**Comment by llamatester** [comment]
The thetis-market-aggregator adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators thetis-market-aggregator
    
    🦙 Running THETIS-MARKET-AGGREGATOR adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 07 Dec 2024 00:00:00 GMT
    End Date:	Sun, 08 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 6.48 M
    End timestamp: 1733529600 (2024-12-07T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs thetis-market
    
    🦙 Running THETIS-MARKET adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 07 Dec 2024 00:00:00 GMT
    End Date:	Sun, 08 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    Version -> THETIS-MARKET-DERIVATIVE
    ---------
    APTOS 👇
    Backfill start time: 26/11/2024
    Daily volume: 5.25 M
    End timestamp: 1733529600 (2024-12-07T00:00:00.000Z)
    
    
    Version -> THETIS-MARKET-AGGREGATOR
    ---------
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 6.48 M
    End timestamp: 1733529600 (2024-12-07T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees thetis-market.ts
    
    🦙 Running THETIS-MARKET.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Sat, 07 Dec 2024 00:00:00 GMT
    End Date:	Sun, 08 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 26/11/2024
    Total fees: 6.03 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Total supply side revenue: 4.82 k
    └─ Methodology: SupplySideRevenue is 80% of the total fees, which are distributed to LP stakers
    Total revenue: 1.21 k
    └─ Methodology: Revenue is 20% of the total fees
    Daily fees: 2.54 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Daily supply side revenue: 2.03 k
    └─ Methodology: SupplySideRevenue is 80% of the total fees, which are distributed to LP stakers
    Daily revenue: 509
    └─ Methodology: Revenue is 20% of the total fees
    End timestamp: 1733529600 (2024-12-07T00:00:00.000Z)

File: aggregators/thetis-market-aggregator/index.ts
Comment (g1nt0ki) [reply aggregators/thetis-market-aggregator/index.ts:1]: hmm, then I would suggest three listings/adapter files. One volume adaptereach  for derivatives,  dex (spot) & and dex aggregator. same for fees.


Also, we have only one tvl thethis adapter listed for perp product, can you put one up for the spot markets?

Comment (thetis-market) [reply aggregators/thetis-market-aggregator/index.ts:1]: Sorry for the confusion in the provided information. Thetis currently offers two main products: Thetis Perpetual Exchange and Thetis Aggregator, similar to Jupiter in the Solana ecosystem.

Thetis Perpetual Exchange: Tracks volume, fees, and TVL.
Thetis Aggregator: Tracks only volume, as it currently has zero fees.

Comment (thetis-market) [reply aggregators/thetis-market-aggregator/index.ts:1]: Hi @g1nt0ki, please review this PR. Thank you!

**Comment by llamatester** [comment]
The thetis-market adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators thetis-market
    
    🦙 Running THETIS-MARKET adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 08 Dec 2024 00:00:00 GMT
    End Date:	Mon, 09 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 3.60 M
    End timestamp: 1733616000 (2024-12-08T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs thetis-market
    
    🦙 Running THETIS-MARKET adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 08 Dec 2024 00:00:00 GMT
    End Date:	Mon, 09 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 9/8/2024
    Daily volume: 3.84 M
    End timestamp: 1733616000 (2024-12-08T00:00:00.000Z)

**Comment by llamatester** [comment]
The thetis-market.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees thetis-market.ts
    
    🦙 Running THETIS-MARKET.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 08 Dec 2024 00:00:00 GMT
    End Date:	Mon, 09 Dec 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 26/11/2024
    Total fees: 7.59 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Total supply side revenue: 6.07 k
    └─ Methodology: SupplySideRevenue is 80% of the total fees, which are distributed to LP stakers
    Total revenue: 1.52 k
    └─ Methodology: Revenue is 20% of the total fees
    Daily fees: 1.56 k
    └─ Methodology: All fees for adding/removing liquidity, margin, liquidation, and swaps are collected
    Daily supply side revenue: 1.25 k
    └─ Methodology: SupplySideRevenue is 80% of the total fees, which are distributed to LP stakers
    Daily revenue: 312
    └─ Methodology: Revenue is 20% of the total fees
    End timestamp: 1733616000 (2024-12-08T00:00:00.000Z)

**Pull request merged**